src/bootstrap/compile.rs

```diff
@@ -474,17 +475,19 @@ pub fn tool(build: &Build, stage: u32, target: &str, tool: &str) {
         cargo.env("LIBZ_SYS_STATIC", "1");
     }
 
-    cargo.env("CFG_RELEASE_CHANNEL", &build.config.channel);
+    if stage == 2 {
```

Comment:
Because it's what the addition intended to be: adding version hash to stage2 tools. I have tested this and it just works without any problem.